components/dash-core-components/src/components/Checklist.react.js

```diff
@@ -33,68 +34,69 @@ export default class Checklist extends Component {
+                                className={labelClassName}
+                            >
+                                <input
+                                    checked={includes(option.value, value)}
+                                    className={inputClassName}
+                                    disabled={Boolean(option.disabled)}
+                                    style={inputStyle}
+                                    type="checkbox"
+                                    onChange={() => {
+                                        let newValue;
+                                        if (includes(option.value, value)) {
+                                            newValue = without([option.value], value);
+                                        } else {
+                                            newValue = append(option.value, value);
+                                        }
+                                        setProps({value: newValue});
+                                    }}
+                                />
+                                {option.label}
+                            </label>
+                })}
             </div>
         );
     }
 }
 
 Checklist.propTypes = {
-    /**
-     * The ID of this component, used to identify dash components
-     * in callbacks. The ID needs to be unique across all of the
-     * components in an app.
-     */
-    id: PropTypes.string,
-
     /**
      * An array of options
      */
     options: PropTypes.arrayOf(
-        PropTypes.exact({
-            /**
-             * The checkbox's label
-             */
-            label: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
-                .isRequired,
-
-            /**
-             * The value of the checkbox. This value
-             * corresponds to the items specified in the
-             * `value` property.
-             */
-            value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
-                .isRequired,
-
-            /**
-             * If true, this checkbox is disabled and can't be clicked on.
-             */
-            disabled: PropTypes.bool,
-        })
+        PropTypes.oneOfType([
+            PropTypes.exact({
+                /**
+                 * The checkbox's label
+                 */
+                label: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
+                    .isRequired,
+
+                /**
+                 * The value of the checkbox. This value
+                 * corresponds to the items specified in the
+                 * `value` property.
+                 */
+                value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
+                    .isRequired,
+
+                /**
+                 * If true, this checkbox is disabled and can't be clicked on.
+                 */
+                disabled: PropTypes.bool,
+            }),
+            PropTypes.string
```

Comment:
Let's add a comment here describing how this is used as label and value, and if the user wants a different thing for label and value then they should use a list instead.. These comments make their way into the Python docstrings in `dcc.Dropdown` via the component generator.